### PR TITLE
Migrate to Capacitor 4

### PR DESCRIPTION
**Pull request by Nevvulo**
Updates the plugin to support Capacitor 4.
I've confirmed this is working for Android and iOS, tested in the simulator and through `npm run verify`

## Capacitor 3 vs Capacitor 4

- Version 0.2.0 (and lower) will target the latest Capacitor version
- Version 1.0.0 will target Capacitor 4

## Breaking Changes
* Plugin must be imported like `import { FirebaseDynamicLinks } from '@joinflux/capacitor-firebase-dynamic-links'` (instead of `CapacitorFirebaseDynamicLinks`)
* Version 1.x.x will solely support Capacitor 4
* No functionality has changed

### CODE REVIEW

**Review by anonimitoraf — COMMENTED** [review]
Cool, doesn't seem like there are logical changes (i.e. non-auto-generated code). I have the same q as segment's: why was the package name updated and whether this has downstream impact?

**Review by anonimitoraf — APPROVED** [review]

File: android/build.gradle [hunk 1]
```diff
@@ -28,21 +35,24 @@ android {
     lintOptions {
         abortOnError false
     }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_11
+        targetCompatibility JavaVersion.VERSION_11
+    }
```
Comment (gugahoi) [inline android/build.gradle:41]: Do we need these?

File: rollup.config.js [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
-  plugins: [
-    nodeResolve()
-  ]
-};
\ No newline at end of file
+  output: [
+    {
+      file: 'dist/plugin.js',
+      format: 'iife',
+      name: 'capacitorCapacitorSegment',
```
Comment (gugahoi) [inline rollup.config.js]: This doesn't seem correct...

**Review by gugahoi — CHANGES_REQUESTED** [review]
I think you missed 1 place when copy pasting.

File: rollup.config.js (hunk 2 above)
Comment (Nevvulo) [reply rollup.config.js]: Thanks, will fix this

**Review by gugahoi — APPROVED** [review]

**Pull request merged**